test/unit/disposable_common.dart

```diff
@@ -272,12 +270,179 @@ void testCommonDisposable(Func<StubDisposable> disposableFactory) {
+        'should call error handler if stream receives an error '
+        'and there was an onError handler set on the subscription', () {
+      // ignore: close_sinks
+      var controller = new StreamController<Null>();
+      // ignore: cancel_subscriptions
+      var subscription = disposable.listenToStream(controller.stream, (_) {});
+      subscription.onError(expectAsync2((error, _) {
+        expect(error, new isInstanceOf<Exception>());
+        expect(error.toString(), 'Exception: intentional');
+      }));
+      controller.addError(new Exception('intentional'));
+    });
+
+    test(
+        'should call onDone callback when controller is closed '
+        'and there was an onDone callback set by listenToStream', () {
+      // ignore: close_sinks
+      var controller = new StreamController<Null>();
+      // ignore: cancel_subscriptions
+      var subscription = disposable.listenToStream(controller.stream, (_) {},
+          onDone: expectAsync0(() {}));
+      controller.close();
+    });
+
+    test(
+        'should call onDone callback when controller is closed '
+        'and there was an onDone callback set on the subscription', () {
+      // ignore: close_sinks
+      var controller = new StreamController<Null>();
+      // ignore: cancel_subscriptions
+      var subscription = disposable.listenToStream(controller.stream, (_) {});
+      subscription.onDone(expectAsync0(() {}));
+      controller.close();
+    });
+
+    test('should un-manage subscription when controller is closed', () async {
+      var previousTreeSize = disposable.disposalTreeSize;
+
+      // ignore: close_sinks
+      var controller = new StreamController<Null>();
+      // ignore: cancel_subscriptions
+      var subscription = disposable.listenToStream(controller.stream, (_) {});
+      expect(disposable.disposalTreeSize, equals(previousTreeSize + 1));
+
+      await controller.close();
+      await new Future(() {});
+
+      expect(disposable.isDisposed, isFalse);
+      expect(disposable.disposalTreeSize, equals(previousTreeSize));
+    });
+
+    test(
+        'should un-manage subscription when controller is closed '
+        'due to an error, when cancelOnError is true', () async {
+      var previousTreeSize = disposable.disposalTreeSize;
+
+      // ignore: close_sinks
+      var controller = new StreamController<Null>();
+      // ignore: cancel_subscriptions
+      var subscription = disposable.listenToStream(controller.stream, (_) {},
+          cancelOnError: true, onError: (_, [__]) {});
+      expect(disposable.disposalTreeSize, equals(previousTreeSize + 1));
+
+      controller.addError(new Exception('intentional'));
+      await new Future(() {});
+
+      expect(disposable.isDisposed, isFalse);
+      expect(disposable.disposalTreeSize, equals(previousTreeSize));
+    });
+
+    test(
+        'should not un-manage subscription when controller is closed '
+        'due to an error, even when cancelOnError is false', () async {
```

Comment:
"should not un-manage subscription when ~~controller is closed due to an error~~ an error is added to the controller, even when cancelOnError is false"

When `cancelOnError` is false, the controller will not close in response to the error.